Tools/MonoGame.Generator.CTypes/PInvokeWritter.cs

```diff
@@ -0,0 +1,150 @@
+// MonoGame - Copyright (C) The MonoGame Team
+// This file is subject to the terms and conditions defined in
+// file 'LICENSE.txt', which is part of this source code package.
+
+using System.ComponentModel;
+using System.Reflection;
+using System.Reflection.Metadata;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Runtime.CompilerServices;
+
+namespace MonoGame.Generator.CTypes;
+
+class PinvokeWritter
+{
+    private readonly string _name;
+
+    private readonly StringBuilder _outputText;
+
+    private readonly Dictionary<string, Type> _handles;
+
+    private readonly List<MethodInfo> _methods;
+
+    private EnumWritter _enumWritter;
+
+    private StructWritter _structWritter;
+
+    public PinvokeWritter(Type type, StructWritter structWritter, EnumWritter enumWritter)
+    {
+        _name = type.Name;
+        _enumWritter = enumWritter;
+        _structWritter = structWritter;
+
+        _outputText = new StringBuilder($"""
+        //
+        // This code is auto generated, don't modify it by hand.
+        // To regenerate it run: Tools/MonoGame.Generator.CTypes
+        //
+
+        #pragma once
+
+        #include "csharp_common.h"
+        #include "csharp_enums.h"
+        #include "csharp_structs.h"
+        
+
+        
+        """);
+
+        _handles = [];
+        _methods = [];
+    }
+
+    private static bool IsHandle(Type type)
+    {
+        if (!type.IsPointer)
+            return false;
+
+        type = type.GetElementType();
+
+        if (type.FullName.StartsWith("System."))
+            return false;
+
+        // Our handle types don't need to be exported
+        // and these are all readonly structs.
+        //
+        // So skip these.
+        //
+        if (type.GetCustomAttribute<IsReadOnlyAttribute>() == null)
```

Comment:
Feels sloppy depending on the `readonly` flag to identify our handle types.

Any better ideas?

Reply:
My idea was to skip the types which names end in `Ptr`.

Reply:
Perhaps a fully qualified attribute? Say annotate with `[MGHandleAttribute]`  which works well with the reflection here as you have the `type`.

Reply:
Yep this worked better:

```
[MGHandle] internal readonly struct MGG_GraphicsSystem { }
```

Thanks @Mindfulplays